pennylane/drawer/tape_text.py

```diff
@@ -177,6 +186,58 @@ def tape_text(
       1: ─├QFT──RY─╰C─┤ ╰Var[Z@Z] ├Probs
       2: ─╰QFT──RZ────┤           ╰Probs
 
+    Matrix valued parameters are always denoted by `M` followed by an integer corresponding to
+    unique matrices.  The list of unique matrices can be printed at the end of the diagram by
+    selecting ``show_matrices=True``:
```

Comment:
I wonder if having so many matrices cached that the integer has double digits will throw off the alignment? Might be worth a quick check?

Reply:
We already have variable length labels. The exact contents of the label doesn't matter, since we just pad to the length of the longest label. I can make a test for this to make doubly sure, but I don't think it's really necessary.